enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/FileUserRealm.java

```diff
@@ -217,6 +218,7 @@ void addUserToRole( String username, String roleName ) throws IOException
                 }
             }
         }
+        clearCachedAuthorizationInfoForUser( username );
     }
 
     void removeUserFromRole( String username, String roleName ) throws IOException
```

Comment:
The thinking was that removeUserFromAllRoles is private so as long as all exposed methods that use it clears the cache that should be enough